cmd/siklu/bank_management.c

```diff
@@ -69,4 +69,53 @@ struct software_bank_t* bank_management_get_current_bank(void) {
 	create_bank_management_info_for_bank(bank);
 	
 	return bank;
-}
\ No newline at end of file
+}
+
+struct software_bank_t* bank_management_handle_auto_switch(struct software_bank_t *bank) {
+	u_char *fdt = NULL;
+	int8_t boot_tries_left = -1; // negative is disabled
+	const char *config_boot_tries_left;
+	bool do_fdt_write = false;
+
+	fdt = siklu_read_fdt_from_mtd_part(CONFIG_SIKLU_BANK_MGMT_MTD_PART);
+	if (! fdt) {
+		printk(KERN_ERR "Could not read bank management info from \"%s\"\n",
+			   CONFIG_SIKLU_BANK_MGMT_MTD_PART);
+		return bank;
+	}
+
+	config_boot_tries_left = siklu_fdt_getprop_string(fdt, "/", PROP_BOOT_TRIES_LEFT, NULL);
+	if (IS_ERR(config_boot_tries_left)) {
+		printf("SIKLU_BOOT: Could not read boot_tries_left\n");
+		return bank;
+	}
+
+	printf("SIKLU BOOT: boot_tries_left = %s\n", config_boot_tries_left);
+	boot_tries_left = simple_strtol(config_boot_tries_left, NULL, 10);
+	if (boot_tries_left < 0) {
+		// do nothing
+		return bank;
+	} else if (boot_tries_left == 0) {
+		// out of tries, switch bank
+		if (bank != &first_bank && bank != &second_bank) {
+			printf("SIKLU_BOOT: Could not switch bank, wrong bank specified\n");
+		} else {
+			bank = (bank == &first_bank) ? &second_bank : &first_bank;
+			siklu_fdt_setprop_string(fdt, "/", PROP_CURRENT_BANK, bank->bank_label);
+			do_fdt_write = true;
+		}
+	}
+
+	// decrement boot_tries_left for next boot
+	char boot_tries_left_str[5]; // "-123\0"
```

Comment:
If our style requires to declare or variable at the beginning then this should be moved upper.
If not then you can move config_boot_tries_left declaration down to the first usage

Reply:
not relevant anymore